Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/common/taimaba/TaimabaApi.java

```diff
@@ -0,0 +1,374 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import okhttp3.HttpUrl;
+
+import static com.github.adamantcheese.chan.core.site.SiteEndpoints.makeArgument;
+
+public class TaimabaApi extends CommonSite.CommonApi {
+    public TaimabaApi(CommonSite commonSite) {
+        super(commonSite);
+    }
+
+    @Override
+    public void loadThread(JsonReader reader, ChanReaderProcessingQueue queue) throws Exception {
+        reader.beginObject();
+        // Page object
+        while (reader.hasNext()) {
+            String key = reader.nextName();
+            if (key.equals("posts")) {
+                reader.beginArray();
+                // Thread array
+                while (reader.hasNext()) {
+                    // Thread object
+                    readPostObject(reader, queue);
+                }
+                reader.endArray();
+            } else {
+                reader.skipValue();
+            }
+        }
+        reader.endObject();
+    }
+
+    @Override
+    public void loadCatalog(JsonReader reader, ChanReaderProcessingQueue queue) throws Exception {
+        reader.beginArray(); // Array of pages
+
+        while (reader.hasNext()) {
+            reader.beginObject(); // Page object
+
+            while (reader.hasNext()) {
+                if (reader.nextName().equals("threads")) {
+                    reader.beginArray(); // Threads array
+
+                    while (reader.hasNext()) {
+                        readPostObject(reader, queue);
+                    }
+
+                    reader.endArray();
+                } else {
+                    reader.skipValue();
+                }
+            }
+
+            reader.endObject();
+        }
+
+        reader.endArray();
+    }
+
+    @Override
+    public void readPostObject(JsonReader reader, ChanReaderProcessingQueue queue) throws Exception {
+        Post.Builder builder = new Post.Builder();
+        builder.board(queue.getLoadable().board);
+
+        SiteEndpoints endpoints = queue.getLoadable().getSite().endpoints();
+
+        // File
+        String fileId = null;
+        String fileExt = null;
+        int fileWidth = 0;
+        int fileHeight = 0;
+        long fileSize = 0;
+        boolean fileSpoiler = false;
+        String fileName = null;
+
+        /* prevent API parse error
+           resto is not available on opening board overview the first time
+           so, we manually set the opId to 0, builder.op to true and builder.opId to 0 */
+        int opId = 0;
+        builder.op(opId == 0);
+        builder.opId(0);
+
+        String postcom = null;
+
+        List<PostImage> files = new ArrayList<>();
+
+        // Country flag
+        String countryCode = null;
+        String trollCountryCode = null;
+        String countryName = null;
+
+        reader.beginObject();
+        while (reader.hasNext()) {
+            String key = reader.nextName();
+
+            switch (key) {
+				case "no":
```

Comment:
Would be nice if you could reformat this code (you can do that by just pressing Ctrl+Alt+L in Android Studio)